docs/getting-started-guides/mesos-docker.md

```diff
@@ -0,0 +1,244 @@
+- [Docker Engine](https://docs.docker.com/) - container management daemon
+  - On Mac, use [Boot2Docker](http://boot2docker.io/) or [Docker Machine](https://docs.docker.com/machine/install-machine/)
+- [Docker Compose](https://docs.docker.com/compose/install/) - multi-container application orchestration
+- [jq](http://stedolan.github.io/jq/) - command line JSON parser
+
+Optional:
+- [Virtual Box](https://www.virtualbox.org/wiki/Downloads) - x86 hardware virtualizer
+  - Required by Boot2Docker and Docker Machine
+- [etcd](https://github.com/coreos/etcd) - key/value store<br/>
+  - Required to run the integration tests locally (e.g. building without `KUBE_RELEASE_RUN_TESTS=N`)
+
+#### Install on Mac (Homebrew)
+
+It's possible to install all of the above via [Homebrew](http://brew.sh/) on a Mac.
+
+Some steps print instructions for configuring or launching. Make sure each is properly set up before continuing to the next step.
+
+```
+brew install git
+brew install jq
+brew install caskroom/cask/brew-cask
+brew cask install virtualbox
+brew install docker
+brew install boot2docker
+boot2docker init
+boot2docker up
+brew install docker-compose
+brew install etcd
+```
+
+#### Install on Linux
+
+Most of the above are available via apt and yum, but depending on your distribution, you may have to install via other
+means to get the latest versions.
+
+It is recommended to use Ubuntu, simply because it best supports AUFS, used by docker to mount volumes. Alternate file
+systems may not fully support docker-in-docker.
+
+
+#### Boot2Docker Config (Mac)
+
+If on a mac using boot2docker, the following steps will make the docker IPs (in the virtualbox VM) reachable from the
+host machine (mac).
+
+1. Set the VM's host-only network to "promiscuous mode":
+
+    ```
+    boot2docker stop
+    VBoxManage modifyvm boot2docker-vm --nicpromisc2 allow-all
+    boot2docker start
+    ```
+
+    This allows the VM to accept packets that were sent to a different IP.
+
+    Since the host-only network routes traffic between VMs and the host, other VMs will also be able to access the docker
+    IPs, if they have the following route.
+
+1. Route traffic to docker through the boot2docker IP:
+
+    ```
+    sudo route -n add -net 172.17.0.0 $(boot2docker ip)
+    ```
+
+    Since the boot2docker IP can change when the VM is restarted, this route may need to be updated over time.
+    To delete the route later: `sudo route delete 172.17.0.0`
+
+
+### Walkthrough
+
+1. Checkout the Kubernetes source
+
+    ```
+    git clone https://github.com/GoogleCloudPlatform/kubernetes
+    cd kubernetes
+    ```
+
+    By default, that will get you the bleeding edge of master branch.
+    You may want a [release branch](https://github.com/GoogleCloudPlatform/kubernetes/releases) instead,
+    if you have trouble with master.
+
+1. Build Kubernetes-Mesos binaries (cross-compiled)
+
+    ```
+    KUBERNETES_CONTRIB=mesos KUBE_RELEASE_RUN_TESTS=N ./build/release.sh
```

Comment:
This cross-compiles to architecture not needed here. For just linux/amd64:

```bash
build/run.sh hack/build-go.sh
```

After that the binaries are in `_output/dockerized/bin/linux/amd64/`

Officially documented in https://github.com/GoogleCloudPlatform/kubernetes/tree/master/build#key-scripts